frontend/dispatcher.lua

```diff
@@ -0,0 +1,258 @@
+local Event = require("ui/event")
+local Screen = require("device").screen
+local UIManager = require("ui/uimanager")
+local T = require("ffi/util").template
+--local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local _ = require("gettext")
+local logger = require("logger")
+
+local Dispatcher = {}
+--[[--
+contains a list of a dispatchable settings
+each setting contains:
+    category: one of none, toggle, absolutenumber, incrementalnumber, or string.
+    event: what to call.
+    title: for use in ui.
+and optionally
+    min/max: for number
+    values: allowed values for string.
+    conditions: (is document open, etc.)
+--]]--
+local settingsList = {
+    page_jmp = { ["category"]="incrementalnumber", ["event"]="GotoViewRel" , ["title"]=_("Go %1 pages"),["min"]=-50, ["max"]=50},
+    prev_chapter = { ["category"]="none", ["event"]="GotoPrevChapter", ["title"]=_("Previous chapter"), },
+    next_chapter = { ["category"]="none", ["event"]="GotoNextChapter", ["title"]=_("Next chapter"), },
```

Comment:
Sorry, this is a nitpick, but rather too early than too late :)
This hurts my eyes :)

For the sake of readability, if your keys are always that simple, just use them as is:
`next_chapter = { category="none", event="GotoNextChapter", title=_("Next chapter"), },`

And even, if there's required and optional items, just use an array:
```lua
page_jmp =     { "incrementalnumber", "GotoViewRel" ,     _("Go %1 pages"),   -50, 50},
next_chapter = { "none",              "GotoNextChapter",  _("Next chapter"), },
```
You can then, if you really need it to be accessible by keys, build another (module local) table with these values as associated arrays.

Reply:
thanks,

an array makes this easier to read but the code harder to understand, so i am not sure if i will go there.